HacktoberfestProject.Web/Services/TableService.cs

```diff
@@ -18,20 +19,41 @@ public TableService(IUserRepository userRepository)
             _userRepository = userRepository ?? throw new ArgumentNullException(nameof(userRepository));
         }
 
-        public Task<ServiceResponse<IEnumerable<Pr>>> AddPrByUsernameAsync(string username, Pr pr)
+        public async Task<ServiceResponse<Pr>> AddPrByUsernameAsync(string username, string owner, string repositoryName, Pr pr)
         {
-            throw new NotImplementedException();
+            var user = await GetUser(username);
+
+            Repository repository = GetReporitory(user , owner, repositoryName);
+
+            var serviceResponse = new ServiceResponse<Pr>
+            {
+                ServiceResponseStatus = ServiceResponseStatus.Created
+            };
+
+            if (!repository.Prs.All(p => p.PrId == pr.PrId))
```

Comment:
PrId is not a unique identifier as far as I am aware. We would need to check against owner, repository _and_ id I believe.

Reply:
The current check you are seeing here is based on the repository that has already been filtered to owner and repository name.